modules/Bio/EnsEMBL/Variation/Utils/BaseVepPlugin.pm

```diff
@@ -326,6 +326,160 @@ sub check_variant_feature_type {
+
+  Arg[1]     : hash
+  Description: Create variant feature from a variant hash
+  Returntype : Bio::EnsEMBL::Variation::VariationFeature
+  Status     : Experimental
+
+=cut
+
+sub _create_vf {
+  my ($self, $var) = @_;
+
+  my $slice = Bio::EnsEMBL::Slice->new(
+    -seq_region_name  => $var->{chr},
+    -start            => $var->{start},
+    -end              => $var->{end},
+    -strand           => $var->{strand},
+  );
+
+  my $vf = Bio::EnsEMBL::Variation::VariationFeature->new(
+    -variation_name        => $var->{variation_name},
+    -seq_region_name       => $var->{chr},
+    -start                 => $var->{start},
+    -end                   => $var->{end},
+    -slice                 => $slice,
+    -strand                => $var->{strand},
+    -allele_string         => $var->{allele_string},
+    -is_somatic            => $var->{somatic},
+    -clinical_significance => $var->{clin_sig} ? [split /,/, $var->{clin_sig}] : []
+  );
+  return $vf;
+}
+
+=head2 _fetch_cache_vars
+
+  Arg[1]     : $seq_region_name
+  Arg[2]     : $start
+  Arg[3]     : $end
+  Description: Fetch variant features within a given genomic region from VEP
+               cache; supports indexed (faster) and non-indexed (slower) cache
+  Returntype : arrayref of Bio::EnsEMBL::Variation::VariationFeature
+  Status     : Experimental
+
+=cut
+
+sub _fetch_cache_vars {
+  my ($self, $chr, $start, $end) = @_;
+
+  my ($as, $variants);
+  if (defined($as = $self->_get_AnnotationSource('VariationTabix'))) {
+    # code based on AnnotationSource::Cache::VariationTabix
+    my $source_chr = $as->get_source_chr_name($chr);
+    my $tabix_obj = $as->_get_tabix_obj($source_chr);
+    return unless $tabix_obj;
+
+    my $iter = $tabix_obj->query(sprintf("%s:%i-%i", $source_chr, $start - 1, $end + 1));
+    return unless $iter;
+
+    while(my $line = $iter->next) {
+      chomp $line;
+      my $var = $as->parse_variation($line);
+      push @$variants, $self->_create_vf($var);
+    }
+  } elsif (defined($as = $self->_get_AnnotationSource('Variation'))) {
+    warn("Using non-indexed VEP cache is slow; for optimal performance, please use indexed VEP cache\n")
+      unless $self->{slow_warning};
+    $self->{slow_warning} = 1;
+
+    # code based on AnnotationSource::Cache::Variation and AnnotationSource
+    my $cache_region_size = $as->{cache_region_size};
+    my ($source_chr, $min, $max, $seen, @regions) = $as->get_regions_from_coords(
+      $chr, $start, $end, undef, $cache_region_size, $as->up_down_size());
+
+    for my $region (@regions) {
+      my ($c, $s) = @$region;
+
+      my $file = $as->get_dump_file_name(
+        $c,
+        ($s * $cache_region_size) + 1,
+        ($s + 1) * $cache_region_size
+      );
+      next unless -e $file;
+      my $gz = gzopen($file, 'rb');
+
+      my $line;
+      while($gz->gzreadline($line)) {
+        chomp $line;
+        # ignore non-overlapping variants
+        my $var = $as->parse_variation($line);
```

Comment:
the non-indexed cache lines do not contain the chromosome, we should add it manually.

```suggestion
        my $var = $as->parse_variation($line);
        $var->{chr} = $c;
```